tabular/src/autogluon/tabular/trainer/abstract_trainer.py

```diff
@@ -933,9 +934,10 @@ def _train_single(self, X, y, model: AbstractModel, X_val=None, y_val=None, **mo
         Trains model but does not add the trained model to this Trainer.
         Returns trained model object.
         """
-        if 'fit_with_prune' in model_fit_kwargs and model_fit_kwargs['fit_with_prune'] is not None:
+        if model_fit_kwargs.get('fit_with_prune_kwargs', None) is not None:
             # This method trains a multiple copy of models and returns the best performing one
-            model = model.fit_with_prune(X=X, y=y, X_val=X_val, y_val=y_val, **model_fit_kwargs['fit_with_prune'], **model_fit_kwargs)
+            fit_with_prune_kwargs = model_fit_kwargs.pop('fit_with_prune_kwargs')
```

Comment:
careful with popping, unless you really know what you are doing you could create serious bugs.

You may want to first do a shallow copy of the dictionary followed by a pop of the copy.

Reply:
In this particular case I believe there isn't an issue though, just something to be aware of.

Reply:
I popped it on purpose because I didn't want `fit_with_prune_kwargs` to also be in `model_fit_kwargs`.

Reply:
I'm aware, I'm mostly saying that this would have impacted the upstream dictionary of model_fit_kwargs if it wasn't used as a kwarg arg in the method you popped it in. Because it was used as the kwarg arg, there isn't an issue however.